### DISCUSSION TITLE
Different behaviour calling tar_* functions from Rmd

### DISCUSSION THREAD

**Discussion by boshek** [ropensci/targets#360] [Help] [answered]
I am working in a windows 10 environment with CRAN versions of targets (0.2.0) and tarchetypes (0.1.0).

When I run something like:
```r
tar_read(some_target)
```
I get different behaviour depending where I called it. If I run this on the console or from a `.R` script it runs fine and returns what I am expecting.

If, however, I call this from an `.Rmd` I get the following error:
```r
Error: utility functions like tar_read() and tar_progress() require a _targets/ data store produced by tar_make() or similar. 
```

So far deleting the `_targets` directory and re-running as well as a fresh clone of the same code both result in this same differential behaviour.

Calling `targets:::path_store` from both environments returns "_targets". For whatever reason, when called from an `.Rmd` [this line](https://github.com/ropensci/targets/blob/8dbc2cf387be5346a5eae3b69876b12697d6d137/R/utils_assert.R#L205) returns `TRUE`.

Anyone have any ideas what might be happening here?

**Comment by wlandau** [comment, marked as answer]
Sounds like your R Markdown document is running from a different root directory from where you run `tar_make()`. Ideally they should agree. Or, you can run something like `withr::with_dir("/my/targets/project/root/dir/", tar_read(my_target))` when you want to read a target in an Rmd. Alternatively, you can set `knitr::opts_knit$set(root.dir = "/my/targets/project/root/dir/")` in a code chunk at the top of the document.

**Reply by boshek** [reply]
Ah! You are totally right. But what happened was some of the RStudio Rmarkdown settings got monkeyed with behind the scenes. Thanks!
